### PR TITLE
EDA notebook adaptions for new dataset

### PR DESCRIPTION
**Pull request by jasminziegler**
# Description

The datasets created by the new architecture have a different folder structure and different target column names (e.g. "key" is now called "scan_type_id") so that we need changes in some of the functions declared in data_utilities.py.
The goal is that the functions work both with the old and the new format of datasets.

Steps:

- [x] 1. Test new utilities with 17k dataset
- [x] 2. refine new utilities
- [x] 3. adapt all other EDA notebooks

User story https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1657/

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] My Code is review by 1 People

### CODE REVIEW

**Comment by jasminziegler** [comment]
@markus-hinsche could you please give me feedback on my first changes? I added some comments in ReviewNB in the relevant sections. Please ignore the rest of the notebook as of now, it is not "ready for review" yet! THX :)

**Comment by markus-hinsche** [comment]
@jasminziegler I can't see any comments. Can you make sure you submitted your comments?

File: src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb [hunk 1]
```diff
@@ -0,0 +1,1769 @@
+{
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: <p>This image looks slightly stretched (too wide)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/425/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: src/data_utils/dataset_EDA/anon-depthmap-17k/.amlignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+## This file was auto generated by the Azure Machine Learning Studio. Please do not remove.
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/anon-depthmap-17k/.amlignore:1]: Please remove this file, consider adding to .gitignore

File: src/data_utils/dataset_EDA/anon-depthmap-17k/data-depthmap-17k.csv [hunk 3]
```diff
@@ -0,0 +1,1018 @@
+"id","scan_type_id","age","sex","height","weight","muac"
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/anon-depthmap-17k/data-depthmap-17k.csv:1]: I think we should remove this file, we usually don't add data to git, just code

File: src/data_utils/dataset_EDA/data_utilities.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
+logger.handlers.clear()
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/data_utilities.py:11]: This isn't working for me, let's consistently address logging issues for the entire repo in a separate PR

File: src/data_utils/dataset_EDA/data_utilities.py [hunk 5]
```diff
@@ -49,27 +50,34 @@ def _count_rows_per_age_bucket(artifacts):
     return df_out
 
 
-def calculate_code_age_distribution(artifacts: pd.DataFrame):
-    codes = list(artifacts['key'].unique())
+def calculate_code_age_distribution(artifacts: pd.DataFrame, scan_type_colname: str):
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/data_utilities.py:53]: Please also add a return type annotation

File: src/data_utils/dataset_EDA/test_data_utilities_find_outlier_qrcodes.py [hunk 6]
```diff
@@ -27,35 +27,35 @@ def prepare_test_df():
 
 def test_find_outlier_qrcodes_age_min():
     df = prepare_test_df()
-    qrs = find_outlier_qrcodes(df, 'age', '<365/2')
+    qrs = find_outliers(df, 'age', '<365/2', 'qr')
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/test_data_utilities_find_outlier_qrcodes.py:30]: Consider re-writing to 

```suggestion
    qrs = find_outliers(df, 'age', '<365/2', data_id_name='qr')
```

in order to improve readability

File: src/data_utils/dataset_EDA/data_utilities.py [hunk 7]
```diff
@@ -49,27 +50,34 @@ def _count_rows_per_age_bucket(artifacts):
...
-    logger.info('No. of qrcodes: %d', len(qrs))
-    return qrs
+    if data_id_name == 'qr':
+        unique_outliers = outlier_artifacts.drop_duplicates(subset='qrcode', keep='first')
+        outliers = unique_outliers.qrcode.tolist()
+        logger.info('No. of outliers: %d', len(outliers))
+    elif data_id_name == 'id':
+        unique_outliers = outlier_artifacts.drop_duplicates(subset='id', keep='first')
+        outliers = unique_outliers.id.tolist()
+        logger.info('No. of outliers: %d', len(outliers))
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/data_utilities.py:77]: These 2 lines are duplicated, so they could live outside of the if/else statement

File: src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb (hunk 1 above)
Comment (markus-hinsche) [reply src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: <p>Any thoughts?</p>

File: src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb [hunk 8]
```diff
@@ -0,0 +1,1488 @@
...
```
Comment (markus-hinsche) [inline src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: <p>Could we also print <code>depthmap.shape</code>  please?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Welthungerhilfe/cgm-ml/pull/425/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: src/data_utils/dataset_EDA/data_utilities.py (hunk 7 above)
Comment (jasminziegler) [reply src/data_utils/dataset_EDA/data_utilities.py:77]: for pulling out line 76, I would need to create another variable and three more lines of code. Pulling the logging out of the if is a good idea

File: src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb (hunk 1 above)
Comment (jasminziegler) [reply src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: I found that "aspect='equal'" as argument for "imshow()" works!

Comment (markus-hinsche) [reply src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: <p><code>aspect="equal"</code> seems to be the default (so  this is what we used before)</p><p>https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.imshow.html</p><p> </p><p>I am fearing a little bit that this image was stretched while resizing, so the data that EDA analyses might be wrong. </p><p>Is this how this image would look in other EDA notebooks (where we know better that the data is right)?</p><p>Thoughts?</p><p> </p>

Comment (jasminziegler) [reply src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: true, I just noticed the same as you are describing here, "aspect" does actually not work on the new dataset, I just saw some difference on one on the older ones. So the depthmap from the new dataset still looks wide no matter whhich "aspect" value I chose, BUT: there is a difference between them, pls see my screenshots:

- Version 1: aspect = auto
grafik

- Version 2: aaspect= equal
grafik

Comment (jasminziegler) [reply src/data_utils/dataset_EDA/anon-depthmap-17k/EDA_anon-depthmap-17k.ipynb:1]: At what point is the resizing happening?

**Review by jasminziegler — COMMENTED** [review]
selfreview done

**Review by markus-hinsche — APPROVED** [review]
LGTM

**Pull request merged**